data/transactions/logic/assembler.go

```diff
@@ -1216,9 +1225,12 @@ func asmTxField(ops *OpStream, spec *OpSpec, args []string) error {
 	if !ok {
 		return ops.errorf("txn unknown field: %#v", args[0])
 	}
-	_, ok = txnaFieldSpecByField[fs.field]
-	if ok {
-		return ops.errorf("found array field %#v in %s op", args[0], spec.Name)
+	// itxn_field appends to the field if it's an array
+	if spec.Name != "itxn_field" {
```

Comment:
is it a new behavior?

Reply:
Yes, because previously no array fields were allowed (they are all in ApplicationCalls).  I did just remove the if test though, since this asm function is only used for itxn_field, it should simply allow all fields.

The versioning mechanism on fields prevents this from changing behavior on old code.  v5 is always going to reject the array fields.